### PR TITLE
[VSMac] Switch macOS build to net6

### PR DESCRIPTION
**Pull request by nosami**
Since some common types got moved out of the `System` namespace in `net6` (`nfloat` etc), there are now some exceptions that causes the extension to not function correctly in VSMac in some cases.

The only way to resolve these issues are to build the extension against the same set of assemblies as the preview version of VSMac. We've been building against an old version (8.4) for quite some time now. This also implies that this extension will need to use `net6` as the TargetFramework.

### CODE REVIEW

File: Src/VimCore/VimCore.fsproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <OutputType>Library</OutputType>
     <RootNamespace>Vim.Core</RootNamespace>
     <AssemblyName>Vim.Core</AssemblyName>
-    <TargetFramework>net472</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (nosami) [inline Src/VimCore/VimCore.fsproj:6]: Will need to ifdef this change

File: nuget.config [hunk 2]
```diff
@@ -4,5 +4,6 @@
     <clear />
     <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
     <add key="vssdk" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk/nuget/v3/index.json" />
+    <add key="vsmacsdk" value="https://devdiv.pkgs.visualstudio.com/DevDiv/_packaging/VSMac/nuget/v3/index.json" />
```
Comment (nosami) [inline nuget.config:7]: This NuGet source is not yet public but `Microsoft.VisualStudioMac.Sdk` should be available from `nuget.org` soon.

The package contains some of the core assemblies used in VSMac.

File: Src/VimCore/VimCore.fsproj (hunk 1 above)
Comment (jaredpar) [reply Src/VimCore/VimCore.fsproj:6]: Should be able to just multi-target here between `net472` and `net6`

File: Src/VimMac/RelativeLineNumbers/CocoaExtensions.cs [hunk 3]
```diff
@@ -11,6 +11,7 @@
 using CoreGraphics;
 using CoreText;
 using Foundation;
+using ObjCRuntime;
```
Comment (jaredpar) [inline Src/VimMac/RelativeLineNumbers/CocoaExtensions.cs:14]: Why is this necessary? Did types move around to a new namespace?

Comment (nosami) [reply Src/VimMac/RelativeLineNumbers/CocoaExtensions.cs:14]: Yes. System.nfloat got moved to ObjCRuntime.nfloat

File: VsVim.sln [hunk 4]
```diff
@@ -30,7 +30,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Documentation", "Documentat
 		Documentation\Supported Features.md = Documentation\Supported Features.md
 	EndProjectSection
 EndProject
-Project("{6EC3EE1D-3C4E-46DD-8F32-0CC8E7565705}") = "VimCore", "Src\VimCore\VimCore.fsproj", "{333D15E0-96F8-4B87-8B03-467220EED275}"
+Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "VimCore", "Src\VimCore\VimCore.fsproj", "{333D15E0-96F8-4B87-8B03-467220EED275}"
```
Comment (nosami) [inline VsVim.sln:33]: Not sure why VSMac changed the project type guid here. I probably should just revert this change.

Comment (jaredpar) [reply VsVim.sln]: Yeah unsure why it did this. SLN files are mostly a mystery to me though.

Comment (nosami) [reply VsVim.sln]: They are a mystery to everyone I think. Only the machines can understand them :)

File: Src/VimCore/FSharpUtil.fs [hunk 5]
```diff
@@ -584,7 +584,7 @@ module CharUtil =
     let IsTagNameChar x = System.Char.IsLetterOrDigit(x) || x = ':' || x = '.' || x = '_' || x = '-'
     let IsFileNameChar x = IsTagNameChar x
     let IsPrintable x =
-        let category = CharUnicodeInfo.GetUnicodeCategory(x)
+        let category = CharUnicodeInfo.GetUnicodeCategory(x: char)
```
Comment (jaredpar) [inline Src/VimCore/FSharpUtil.fs:587]: Were the type's necessary here and forced by an F# change in inference?

Comment (nosami) [reply Src/VimCore/FSharpUtil.fs:587]: although now I just looked at that code again, it would probably be neater to add the type annotation to `IsPrintable x` instead.

Comment (jaredpar) [reply Src/VimCore/FSharpUtil.fs:587]: Either way is fine. I'm just always curious if it's a bug / broken behavior the code was relying on in F# or if there was a diff reason. Found a few times where I had relied on F# bugs.

**Comment by nosami** [comment]
I think this is pretty much ready now. As far as I can tell/guess, this build will pass as soon as we have a public URL for VSMac preview 7 to build against.

**Review by jaredpar — APPROVED** [review]

**Comment by nosami** [comment]
There are tons of warnings that look like this

```
/Users/jasonimison/src/vsvim/Src/VimMac/VimHost.cs(622,33): warning CA1416: This call site is reachable on all platforms. 'Process.StandardInput' is only supported on: 'macOS/OSX'. [/Users/jasonimison/src/vsvim/Src/VimMac/VimMac.csproj]
```

I can make the warnings go away by targeting `net6.0-macos` instead of `net6` but then it is no longer buildable on Windows. I think I just need to ignore these warnings for now.
